source/extensions/tracers/datadog/time_util.cc

```diff
@@ -0,0 +1,26 @@
+#include "source/extensions/tracers/datadog/time_util.h"
+
+#include <chrono>
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+
+datadog::tracing::TimePoint estimateTime(SystemTime wall) {
```

Comment:
why are we not plumbing the MonotonicTime (or maybe just the TimeSource into here?

Reply:
The goal is to implement the interfaces defined in <https://github.com/envoyproxy/envoy/blob/main/envoy/tracing/trace_driver.h>.

There are two time points in those interfaces: [Span::spawnChild][1] and [Driver::startSpan][2]. Both take `SystemTime` only, without a monotonic counterpart.

If I were redesigning the interface, I would have `spanChild` and `startSpan` take both a `SystemTime` and a `MonotonicTime`, and additionally have `finishSpan` take a `MonotonicTime` (or both, but at least the `MonotonicTime`).

I thought it would be inappropriate to propose to change the interface and everything that implements it.

What do you think?

[1]: https://github.com/envoyproxy/envoy/blob/ae071db29b5e7ceaed4a4b76e6d7b7c138f18f2d/envoy/tracing/trace_driver.h#L58-L65
[2]: https://github.com/envoyproxy/envoy/blob/ae071db29b5e7ceaed4a4b76e6d7b7c138f18f2d/envoy/tracing/trace_driver.h#L110-L112

Reply:
I think it's fine to push forward with this but we should also try to correct the API to get the best results. It's not that hard to change these APIs; there's probably not that much fallout, particularly if you just plumb through some additional fields and don't remove the SystemTime.